### ISSUE TITLE
Kernel upgrade breaks displaylink driver - external monitors no longer detected

### ISSUE THREAD

**Issue by jtmackoy** [pop-os/pop#2027]
**Distribution:**
 ```
NAME="Pop!_OS"
VERSION="21.04"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.04"
VERSION_ID="21.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute
LOGO=distributor-logo-pop-os
```
  
**Related Application and/or Package Version:**
DisplayLink Linux Software 5.4.1-55.174 (either [manually installed from here](https://www.synaptics.com/products/displaylink-graphics/downloads/ubuntu) or via [displaylink-debian](https://github.com/AdnanHodzic/displaylink-debian)).
More specifically, the evdi kernel module needed to support functions of the DisplayLink driver.

**Issue/Bug Description:**
Ran into a problem whereby my Plugable ud-ultc-4k dock stopped displaying two external monitors after an upgrade to 5.15.5-76051505-generic of the pop-os kernel.  Ultimately, the problem was due to evdi not compiling properly when dkms tried to update modules for the new kernel.  [This proved a sufficient workaround for now](https://github.com/DisplayLink/evdi/issues/308#issuecomment-981731578) until the [evdi devs release an updated version sometime in December](https://github.com/DisplayLink/evdi/issues/308#issuecomment-979803934).

**Steps to reproduce:**
1. Update to pop-os kernel 5.15.5-76051505-generic (or later, presumably)
2. Reboot
3. External displays are no longer detected

**Expected behavior:**
1. Update to pop-os kernel 5.15.5-76051505-generic (or later, presumably)
2. Reboot
3. External displays are detected as with the previous kernel

**Other Notes:**
The root of this issue really is with the evdi kernel module not compiling successfully against the new kernel, however; it wasn't abundantly clear to me that was the case until digging further into the problem.

Wondering if the pop-os devs can facilitate a fix of some sort (similar to the workaround linked above) in a supported way, rather than users having to hand-edit source files.

At minimum, hopefully this issue to help others that may be affected by this to find a suitable workaround faster than I did.

**Comment by jacobgkau** [comment]
The release notes for the latest DisplayLink driver only mention support for up to kernel version 5.13. I see in the [issue](https://github.com/DisplayLink/evdi/issues/322) you linked that another user has created a patch to evdi for kernel 5.15 support, and there is also an open PR to add 5.15 support, but it has not been merged by the DisplayLink developers yet: https://github.com/DisplayLink/evdi/pull/327 

> Wondering if the pop-os devs can facilitate a fix of some sort (similar to the workaround linked above) in a supported way, rather than users having to hand-edit source files.

Providing this fix in Pop!_OS before it's in upstream DisplayLink would basically amount to patching, maintaining, and shipping DisplayLink/evdi within Pop!_OS. It's something to consider, since DisplayLink does seem to lack current kernel support on a regular basis, although it would come with a maintenance burden.

**Comment by jtmackoy** [comment]
> It's something to consider, since DisplayLink does seem to lack current kernel support on a regular basis, although it would come with a maintenance burden.

Fair enough.  I may be in the minority in terms of those who use the DisplayLink driver with pop-os, in which case I would absolutely understand not taking on that maintenance burden.

Maybe simply having this issue exist, and crawl-able by various search engines, is good enough.

**Comment by iamarkadyt** [comment]
@jtmackoy you're not a minority. Recently updated to 5.15 and my monitors stopped working as well. Had to downgrade to an older version.

**Comment by lgonzalezsa** [comment]
Same boat here, updated to 5.15 kernel and external monitors not working. Any pointer to how downgrade to 5.13? thanks!

**Comment by lgonzalezsa** [comment]
Was able to run a "patch" from here: https://github.com/DisplayLink/evdi/issues/322#issuecomment-982908544
External monitors working now

**Comment by timnolte** [comment]
This has now been broken yet again. Why is it that Pop!_OS can't maintain DisplayLink reliability? The majority of docking stations out there use DisplayLink and most all other platforms work without it breaking all the time.

**Comment by leviport** [comment]
The DisplayLink driver is not maintained by Pop, or even by Ubuntu. The Linux version of the DisplayLink driver also has a reputation for being unreliable. Unfortunately there's not much the Pop team can do about the DisplayLink driver, other than recommend non-DisplayLink docks. 

That said, it's probably worth checking to make sure the [latest version](https://www.synaptics.com/products/displaylink-graphics/downloads/ubuntu) is installed, just in case DisplayLink has updated it and improved compatibility.

**Comment by timnolte** [comment]
My Chromebook works with DisplayLink no problem and I've never had a ChromeOS update break it. Obviously even though the fact is that DisplayLink is more widely used in docking stations then non-DisplayLink units you apparently aren't concerned with that. I would be sure to state very clearly in all documentation that you have no intention to support DisplayLink and users should expect problems if they attempt to use devices that require it.

**Comment by timnolte** [comment]
This issue, as well as any others related to DisplayLink, should just be immediately closed if/when they are opened with a notice that you don't support DisplayLink.

**Comment by leviport** [comment]
You're misunderstanding the situation. DisplayLink makes the driver. Not Pop and not Ubuntu. _They_ claim to support Ubuntu (and though not stated, Pop by proxy). If they open-sourced it, maybe we could do something about it.

**Comment by jtmackoy** [comment]
I noticed @timnolte (assuming this was him - apologies if not) deleted his comments - wish he hadn't.  As a Pop!-OS user, I share his frustrations with how DisplayLink fits into the integration of my system.  At the same time, I understand the Linux ecosystem is ... different than more tightly curated OSes - like ChromeOS, Windows, or MacOS.  For me, that's both a blessing and a curse: one which I understand sometimes requires more leg work on my part to continue using.

That said ...

Last kernel update also broke my DisplayLink setup.  FWIW, I was able to simply download the latest DisplayLink driver (v5.6, from @leviport's [link above](https://github.com/pop-os/pop/issues/2027#issuecomment-1203201362)), then install it via the [directions also on the DisplayLink site](https://support.displaylink.com/knowledgebase/articles/684649) with no complications.  Previous attempts in the past when this has happened sometimes required manual uninstallation of the existing driver but that was not the case with today's instance.  After a reboot, my DisplayLink hub connected monitors sprang back to life.

**Comment by leviport** [comment]
Since that driver isn't in any repositories or PPAs, it needs to be manually updated. Glad it got yours working again.
